simulate/CMakeLists.txt

```diff
@@ -129,8 +130,13 @@ add_library(mujoco::libsimulate ALIAS libsimulate)
 
 target_sources(
   libsimulate
-  PUBLIC simulate.h
-  PRIVATE simulate.cc array_safety.h
+  PUBLIC 
+    simulate.h
+    platform_ui_adapter.h
+    glfw_adapter.h
+    glfw_dispatch.h
+    array_safety.h
```

Comment:
Do we need all these headers to be public?

Also check also the `target_link_libraries` line. Now that we install the target we need to ensure its public dependencies are also available. Can they be moved to private?

Reply:
Got it